### ISSUE TITLE
Describe > Specific > Boxplot enhancements with general application

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#2084]
I suggest there is an "interesting" general issue here.  And it provides me - I hope - with a simple suggestion.  It is almost certainly post-beta, but just may be quite simple.

My understanding is that themes ( that have yet to be addressed in the graphics work) are everything that is **not** data dependent.  I really like that idea.  

Now what about the orientation of axis labels.  It sounds trivial, but we get the problem a lot, because we often have factors - and do bar charts.  And then the labels overlap and you can see nothing.  

One example, for my climatic work is that we will sometimes have the year as a factor.  I like boxplots going vertically when there are lots of years - so I don't want to do coord_flip.  But I have no clue about the x-axis anymore, because the year labels go into a blur.

The answer is to choose the orientation of the axis label - and make it vertical.

Now this is interesting, because it is a feature of themes in ggplot2, but it isn't fully data independent.  I want it because my factor has a lot of levels.

In most graphics packages this aspect is on the x-axis and y-axis tabs.  I suggest that is an obvious place for it here too.  When we get to looking at themes, then it may come there too - and that would be fine.

So, I suggest that Orientation of axis labels be added to (at least) the x-axis tab.  It is less important for the y-axis, because the labels are at 90 degrees by default.  There it would be the opposite reason, i.e. if there are just a few levels, then they don't need to be at 90 degrees.  

It may be that we just put a simple option here, e.g. just 90 degrees, and leave the full flexibility to the themes. 

The fact that this issue is slightly data dependent and hence qualifies to be slightly there - under the axis labels, could become a good feature in our discussions on the logic of the ggplot2 dialogues.

Is that sufficient reason to investigate the issue for the beta?  Almost certainly not!  Here is one of the suggestions when you search for the issue:

_ggplot(tabla,aes(Indicador,Número))+geom_bar(stat="identity")+
  theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5))_

It is here [http://stackoverflow.com/questions/15629192/making-a-bar-chart-in-ggplot-with-vertical-labels-in-x-axis](url)
 A second link is here [http://stackoverflow.com/questions/1330989/rotating-and-spacing-axis-labels-in-ggplot2](url)

That gives a very nice example with years given vertically.  It looks as though it should be really easy for you graphics whizzes to do - compared with most of your current tasks!

Roger

**Comment by rdstern** [comment]
I changed the name, because I would like to consider a more general aspect of the topic posed above.  This is partly because I am finding the boxplot dialogue is going to be really important for our climatic work.

So I have been looking at alternative solutions to the issue above.  This is partly because there should be more options here, given the "grammar of graphics" approach.  Some could change the boxplot dialogue is specific ways and others are more general issues.

Once someone is ready I suggest this could point to a useful look at this dialogue again, and perhaps discussions could start this week, while Danny is around?

What has led to this discussion is the need for axis labels when the x-axis has many labels, so they all overlap.

1) The suggestion above was to consider adding the orientation of the axis labels to the x-axis (and y-axis sub-dialogues.  I know it is part of themes in ggplot jargon, but it is a case where the theme context is partially mixed with the data, so we could have it perhaps in 2 places.
2) Boxplots can be used when the x-variable is a variate - rather than a factor.  Here is an example from the documentation  ggplot(diamonds, aes(carat, price)) + geom_boxplot(aes(group = cut_width(carat, 0.25)))
So could we permit this (exceptionally in the main dialogue - or somewhere)?  If a variate, then you need to group by the cut_width, so there would be an additional field and the default cut_width could be 1.  I wonder if this could go on the boxplot options?  Then the year could be a variate and I think the axis labels would then be fine.
3) When making the year into a factor it is logically an ordered factor.  I feel there could usefully be an option with an ordered factor to not display all levels on the axis.  I don't know whether this is indeed possible, or whether it makes a good question to ask?

So I would like a range of options to be explored for the boxplot dialogue to solve this issue.  And ideally there would be a range of solutions.

And I think some of the topics here are more general than boxplots.

And while on the subject of boxplots I hope that it will not be too long before that dialogue is corrected and enhanced.  
a) There is still the bug/inconvenience on the facets.
b) Adding a default Jitter layer would be good on the main dialogue.
c) Having the whole dialogue to choose between 1) boxplots and 2) violin (and possibly also 3) Box + Violin and 4) Violin + Box) would be good.  Though adding violin to box and vice versa might be better in the ordinary way as an additional layer.  
d) I wanted to try adding a layer myself, but I don't see **violin** or **jitter** as possible layers.  I assume they will be coming?

**Comment by dannyparsons** [comment]
text angle implemented on themes subdialog, moved other points to new issue
